pkg/ingester/ingester_claim.go

```
@@ -102,13 +106,25 @@ func (i *Ingester) TransferChunks(stream client.Ingester_TransferChunksServer) e
 		return err
 	}
 
+	i.joiningSampleQueueLock.Lock()
 	i.userStatesMtx.Lock()
-	defer i.userStatesMtx.Unlock()
+	i.userStates = userStates
+	i.userStatesMtx.Unlock()
+
+	for j := range i.joiningSampleQueue {
+		userSamples := &i.joiningSampleQueue[j]
+		userCtx := user.InjectOrgID(stream.Context(), userSamples.userID)
+		for k := range userSamples.samples {
+			i.append(userCtx, &userSamples.samples[k])
```

Returned error is ignored.

---

Should I just log the error?